helix-core/src/syntax.rs

```diff
@@ -743,16 +748,22 @@ impl Syntax {
 
                 // If a tree already exists, notify it of changes.
                 if let Some(tree) = &mut layer.tree {
-                    for edit in edits.iter().rev() {
-                        // Apply the edits in reverse.
-                        // If we applied them in order then edit 1 would disrupt the positioning of edit 2.
-                        tree.edit(edit);
+                    if let Some(&modified) = modified.get(&layer_id) {
+                        for edit in edits.iter().rev() {
+                            // Apply the edits in reverse.
+                            // If we applied them in order then edit 1 would disrupt the positioning of edit 2.
+                            tree.edit(edit);
+                        }
+
+                        if modified {
```

Comment:
I'm confused how this works actually since `modified = true` only for the root layer

Reply:
Sorry about that, I previously used two `HashSets` (one for `modified=true`, one for `modified=false`) and after testing refactorered to use a single `HashMap<bool>`. During that redactor I accidently used `entry(..).or_insert()` everwehere. 

Thus works because if the modification is before the start of the layer ( modified is `false`) `tree.edit`is enough to move the injection. Reparsing is only necessary if the content inside the injection actually changed